Runtime/Scripts/Layers/BinaryMeshLayer/BinaryMeshLayer.cs

```diff
@@ -185,6 +182,9 @@ private IEnumerator LoadMetaData(GameObject gameObject, string geometryUrl)
 
         private void ReadMetaDataFile(byte[] results, GameObject gameobject)
         {
+            // The gameobject could be destroyed in the mean time
+            if (!gameobject) return;
```

Comment:
Minor fix #3, because of the other changes we now need to take into consideration the GameObject may be 'destroying' in the same frame; using the ! operator will also check the lifecycle of a gameobject and return true if it technically still exists but is in the process of destroying